File: cpp/src/GadoLeiteiro.cpp
#include "../include/GadoLeiteiro.h"
#include <iostream>

GadoLeiteiro::GadoLeiteiro(const std::string& identificacao, int idade) {
    this->identificacao = identificacao;
    this->idade = idade;
}

int GadoLeiteiro::totalFerteis() {
    return (idade > 1 && idade < 10) ? 1 : 0;
}

void GadoLeiteiro::manejar() {
    std::cout << "Manejo específico das vacas." << std::endl;
    for (const auto& t : tratamentos) {
        std::cout << "- Data: " << t.getData() << ", Tipo: " << t.getTipo() 
                  << ", Produto: " << t.getProduto() << std::endl;
    }
    std::cout << std::endl;
}

File: python/gado_leiteiro.py
from animal import Animal

class GadoLeiteiro(Animal):
    def __init__(self, identificacao, idade):
        super().__init__()
        self.identificacao = identificacao
        self.idade = idade

    def total_ferteis(self):
        if 1 < self.idade < 10:
            return 1
        else:
            return 0

    def manejar(self):
        print("Manejo específico das vacas.")
        for t in self.tratamentos:
            print(f"- Data: {t.get_data()}, Tipo: {t.get_tipo()}, Produto: {t.get_produto()}")
        print()